src/Steam/Provider.php

```diff
@@ -140,11 +146,17 @@ private function buildUrl()
     {
         $realm = $this->getConfig('realm', $this->request->server('HTTP_HOST'));
 
+        if ($this->request->secure()) {
+            $realm = sprintf('https://%s', $realm);
+        } else {
+            $realm = sprintf('http://%s', $realm);
+        }
+
         $params = [
             'openid.ns'         => self::OPENID_NS,
             'openid.mode'       => 'checkid_setup',
             'openid.return_to'  => $this->redirectUrl,
-            'openid.realm'      => sprintf('https://%s', $realm),
+            'openid.realm'      => $realm,
```

Comment:
```suggestion
            'openid.realm'      => sprintf('%s://%s', $this->request->getScheme(), $realm),
```

Save using the conditional :)

Reply:
thanks!